net/zmap/Portfile

```diff
@@ -16,25 +16,30 @@ long_description    designed for Internet-wide network surveys capable of \
                     a 10GbE connection.
 homepage            https://zmap.io/
 
-checksums           rmd160  4b30c0f33c5e4503a89dcb37808c50075373c4f4 \
-                    sha256  61787bd59cb23183a8b5424a148b95812d8254d6249c2d4efb31cbebff9ca383 \
-                    size    154851
+checksums           rmd160  557341e8a872eb61f8c91b64c5910c024fb7efac \
+                    sha256  35d2783b4701549caa0f099d661ba46ad96627ccf00d7352f4ee61a4702a78d8 \
+                    size    156261
 
 depends_build-append port:byacc \
                     port:flex \
                     port:gengetopt \
                     port:pkgconfig \
-                    port:cmake
+                    port:cmake \
+                    port:json-c
 
 depends_lib-append  port:gmp \
                     port:libdnet \
                     port:libpcap \
-                    port:json-c \
-                    port:libunistring \
+                    port:libunistring
 
+default_variants +release
 
 variant redis description {Add Redis support} {
     depends_lib-append port:redis
     depends_lib-append port:hiredis
     configure.args-append -DWITH_REDIS=ON
 }
+
+variant release description {turns off debug symbols and performance impacting defaults } {
+    configure.args-append -DENABLE_DEVELOPMENT=OFF -DENABLE_LOG_TRACE=OF
+}
```

Comment:
The cmake portgroup defines a debug variant. Ports should be coded so that you get the release version if you do not enable that variant. For most ports, this happens automatically with no added code, but I'm not familiar with how zmap's CMakeLists.txt is coded.

Reply:
That is a useful insight.

I added this after noticing the Development Notes here: https://github.com/zmap/zmap/blob/main/INSTALL.md

Though the ZMAp CMakeLists.txt can be gleaned here: https://github.com/zmap/zmap/blob/main/CMakeLists.txt

As it currently stands (for RC2 at least):

```
set(ZMAP_VERSION DEVELOPMENT) # Change DEVELOPMENT to version number for release
```
 
May have had me misinterpreting that, especially since

`option(ENABLE_DEVELOPMENT "Enable development specific compiler and linker flags" OFF)`

Also seems to default to OFF

Though, as I re-read this, it seems as if I had a typo:

` -DENABLE_LOG_TRACE=OF
`

Which should have read

` -DENABLE_LOG_TRACE=OFF`

Regardless, the CMakeLists.txt also appears to default to OFF:

`option(ENABLE_LOG_TRACE "Enable log trace messages" OFF)
`

So upon further scrutiny with your perspective, it seems entirely likely this variant is not necessary and I was overdoing it with a misinterpretation of the Development Notes from INSTALL.md and not understanding the cmake portgroup nuances enough relative to CMakeLists.txt.

Having written as much, I see that the other changes which you recommended that I committed, appear to have broken on the build bots as I am typing this!

My *guess* is it will have to do with the checksums and the now more precisely defined github.tarball_from archive causing checksum mismatches, but I need to dig into those logs and verify my suspicions before jumping ahead of myself.